### ISSUE TITLE
Long pooling socket time out exception

### ISSUE THREAD

**Issue by abangkis** [pengrad/java-telegram-bot-api#36]
Hi, I'm using long pooling to get the updates instead of webhook. When there's no message for the bot, it throws SocketTimeOutException and it's clutering server log, since i run the updates every minute.        doing  catch (Exception e) {} seems to be a bad practice, is there any better way to handle this? 

Thanks.

**Comment by pengrad** [comment]
Hi, thanks for response. 
I dont think its really bad practice to catch SocketTimeOutException, cause it can occur not only in getUpdates. But in your case it is comfortable to avoid it. I will think about some optional reconnection...

Right now you can: 
1) Do not print exception stacktrace in log :)
2) Set timeouts to custom OkHttpClient (http://stackoverflow.com/questions/25953819/how-to-set-connection-timeout-with-okhttp) and build bot with it like `TelegramBotAdapter.buildCustom(token, httpclient)`

**Comment by localhoster** [comment]
Thank you @pengrad, i had the same issue and just catched the exception to  "/dev/null"every time:)
Now the second solution works fine for me. No more exceptions:
```
OkHttpClient client = new OkHttpClient.Builder()
	        .connectTimeout(15, TimeUnit.SECONDS)
	        .writeTimeout(15, TimeUnit.SECONDS)
	        .readTimeout(60, TimeUnit.SECONDS)
	        .build();
		TelegramBot bot = TelegramBotAdapter.buildCustom(Global.token,client);
```
....
```
GetUpdatesResponse updatesResponse = bot.execute(
				    new GetUpdates().limit(10).offset(updateId).timeout(10000));
```

**Comment by pengrad** [comment]
Fixed in 2.1.2
Read timeout for GetUpdates is longer then request's timeout.

**Comment by sagiv562** [comment]
@pengrad What is the meaning of GetUpdates.timeout.
After something like 2 hours that my bot is running i get "timeout exception", do i need to deploy the bot or something ?
